r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -711,3 +711,71 @@ test_that("am/pm mirror lubridate", {
   )
 
 })
+
+test_that("extract tz", {
+  df <- tibble(
+    x = as.POSIXct(c("2022-02-07", "2022-02-10"), tz = "Pacific/Marquesas"),
+    #lubridate::tz() returns -for the time being - "UTC" for NAs, strings,
+    #dates and numerics
+    y = c("2022-02-07", NA),
+    z = as.Date(c("2022-02-07", NA)),
+    w = c(1L, 5L),
+    v = c(1.1, 2.47)
+  )
+
+  compare_dplyr_binding(
+    .input %>%
+      mutate(timezone_x = tz(x)) %>%
+      collect(),
+    df
+  )
+
+  expect_snapshot(
+    compare_dplyr_binding(
+      .input %>%
+        mutate(
+          timezone_y = tz(x),
+          timezone_z = tz(y)
+        ) %>%
+        collect(),
+      df
+    ),
+    error = TRUE
```

Comment:
I think what you're looking for here (and below too) is:

```suggestion
    warning = TRUE
```


That will ensure that the warning we're getting is the "not supported" kind, and then compare the results after that.

https://github.com/apache/arrow/blob/96785665eff453aa4e5fc87a8ee5d047b9526869/r/tests/testthat/helper-expectation.R#L81-L85

`error = TRUE` is being passed via `...` to `expect_equal()` which also has `...` which ends up being silently ignored.

Reply:
Not sure what to say:
* `error` is an argument to `expect_snapshot()` and `warning` is to `compare_dplyr_equal()` 
* I like the `expect_shapshot(..., error = TRUE)` a bit better since with the change we would lose some of the message cover:
<img width="1199" alt="image" src="https://user-images.githubusercontent.com/13176361/154141559-aa613f56-e790-4ae2-af5b-488bd8ce43ee.png">

Reply:
OH, sorry I was off one level of the nesting. I might be misunderstanding the output that you're seeing there, but I interpret that as the snapshot is recording that there's no output (i.e. the expectation has passed). Which is what we expect with `warning = TRUE`

Reply:
We could use `warning = TRUE` and `compare_dplyr_binding()` (and no longer use  `expect_snapshot()`), but we will no longer be testing the error messages (which we worked to improve only because they were visible).

Reply:
I would say that if we want to check the error messages, we should do that directly with `expect_error()` or `expect_snapshot()` wrapped around something like: 

```
call_binding("tz", "this is a string!")
```

Reply:
What I was testing was the `warning` in `compare_dplyr_warning()` being promoted to `error` level. This is the error we improved which otherwise would not have been visible.